auth/realm/base/src/main/java/org/wildfly/security/auth/realm/DistributedSecurityRealm.java

```diff
@@ -183,7 +195,19 @@ private boolean nextIdentity() throws RealmUnavailableException {
             }
             currentIdentity = securityRealms[nextRealm].getRealmIdentity(principal);
             nextRealm++;
```

Comment:
I think we need to be careful with where this is being incremented. If `getRealmIdentity` throws an exception, this won't be incremented and the wrong value will be passed to the `callback.accept` call. So it would be good to check this when updating the `try/catch`.